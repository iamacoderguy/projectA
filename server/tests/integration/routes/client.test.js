const request = require('supertest');
let server;

describe('/', () => {
    beforeEach(() => { server = require('../../../index'); });
    afterEach(() => { server.close(); });

    describe('GET /', () => {
        it('should return the hello world message', async () => {
            const res = await request(server).get('/');
            
            expect(res.status).toBe(200);
            expect(res.text).toMatch(/hello/i);
            expect(res.text).toMatch(/world/i);
        });
    });
});