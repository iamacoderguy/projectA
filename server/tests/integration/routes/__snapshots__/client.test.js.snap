// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/ GET / when client is on localhost should render adminDashboardView correctly in case request ip = ::1 1`] = `"<!DOCTYPE html><html lang=\\"en-US\\"><head><title>Server-A Dashboard</title><meta charset=\\"utf-8\\"><meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\"><meta name=\\"description\\" content=\\"Server-A Dashboard\\"><link rel=\\"stylesheet\\" href=\\"w3.css\\"></head><body><div class=\\"w3-container\\"><div class=\\"w3-container w3-dark-teal\\"><h2>Server-A Dashboard - 192.168.5.1:3000</h2></div><p><iframe width=\\"100%\\" height=\\"200vw\\" title=\\"Result box\\" srcdoc=\\"&lt;i&gt;Result...&lt;/i&gt;\\" name=\\"iframe_result\\"></iframe></p><form class=\\"w3-container\\" method=\\"POST\\" action=\\"/api/files/path?_method=PUT\\" target=\\"iframe_result\\"><p><label class=\\"w3-text-dark-teal\\" for=\\"sharedPathInput\\"><b>Set a new shared path (/api/files/path)</b></label><input class=\\"w3-input w3-border w3-round\\" id=\\"sharedPathInput\\" name=\\"path\\" type=\\"text\\" placeholder=\\"path-to-files\\" value=\\"path-to-files\\"></p><p><button class=\\"w3-btn w3-dark-teal\\" type=\\"submit\\">PUT</button></p></form><ul class=\\"w3-ul\\"><li><h2>List of clients:</h2></li><li class=\\"w3-bar\\"><div class=\\"w3-bar-item\\">111.222.333.444: connecting</div></li><li class=\\"w3-bar\\"><div class=\\"w3-bar-item\\">111.222.333.555: disconnected</div></li><li class=\\"w3-bar\\"><div class=\\"w3-bar-item\\">111.222.333.666: connecting</div></li></ul></div></body></html>"`;

exports[`/ GET / when client is on localhost should render adminDashboardView correctly in case request ip = 127.0.0.1 1`] = `"<!DOCTYPE html><html lang=\\"en-US\\"><head><title>Server-A Dashboard</title><meta charset=\\"utf-8\\"><meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\"><meta name=\\"description\\" content=\\"Server-A Dashboard\\"><link rel=\\"stylesheet\\" href=\\"w3.css\\"></head><body><div class=\\"w3-container\\"><div class=\\"w3-container w3-dark-teal\\"><h2>Server-A Dashboard - 192.168.5.1:3000</h2></div><p><iframe width=\\"100%\\" height=\\"200vw\\" title=\\"Result box\\" srcdoc=\\"&lt;i&gt;Result...&lt;/i&gt;\\" name=\\"iframe_result\\"></iframe></p><form class=\\"w3-container\\" method=\\"POST\\" action=\\"/api/files/path?_method=PUT\\" target=\\"iframe_result\\"><p><label class=\\"w3-text-dark-teal\\" for=\\"sharedPathInput\\"><b>Set a new shared path (/api/files/path)</b></label><input class=\\"w3-input w3-border w3-round\\" id=\\"sharedPathInput\\" name=\\"path\\" type=\\"text\\" placeholder=\\"path-to-files\\" value=\\"path-to-files\\"></p><p><button class=\\"w3-btn w3-dark-teal\\" type=\\"submit\\">PUT</button></p></form><ul class=\\"w3-ul\\"><li><h2>List of clients:</h2></li><li class=\\"w3-bar\\"><div class=\\"w3-bar-item\\">111.222.333.444: connecting</div></li><li class=\\"w3-bar\\"><div class=\\"w3-bar-item\\">111.222.333.555: disconnected</div></li><li class=\\"w3-bar\\"><div class=\\"w3-bar-item\\">111.222.333.666: connecting</div></li></ul></div></body></html>"`;
